odxtools/diagcodedtypes.py

```diff
@@ -547,18 +547,14 @@ def convert_internal_to_bytes(self, internal_value, encode_state: EncodeState,
     def convert_bytes_to_internal(self, decode_state: DecodeState, bit_position: int = 0):
         # Find length key with matching ID.
         bit_length = 0
-        for parameter, value in decode_state.parameter_value_pairs:
-            # if isinstance(param_value.parameter, LengthKeyParameter) would be prettier,
-            # but leads to cyclic import...
-            if (parameter.parameter_type == "LENGTH-KEY" and
-                    parameter.odx_id == self.length_key.odx_id  # type: ignore[attr-defined]
-               ):
+        for parameter_name, value in decode_state.parameter_values.items():
+            if parameter_name == self.length_key.short_name:
```

Comment:
does this logic assume the length and the value are in the same struct? or it can handle length and value belonging to different parameters in the hierarchy?

Reply:
as far as I can see, it is the latter as long as the length key is physically in front of the value. note that using IDs does not change these aspects of the coding logic.